### PR TITLE
Doc for Edge APIs usage

### PR DESCRIPTION
**Pull request by emdobrin**
## Description

Provides examples for using the Edge public APIs for application and extension developers.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/ExtensionUsage.md [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+ExperienceEvent experienceEvent = new ExperienceEvent.Builder()
+	.setXdmSchema(xdmData)
+	.build();
+```
+
+
+
+**Create Experience Event from XDM Schema implementations:**
+
+The AEP Edge extension provides an XDM Schema interface / protocol for you to create Classes for XDM Schema representations. In the example below the `XDMSchemaExample` is a representation of an XDM Schema originating from an Experience event, with two fields: `eventType` and `otherField`. It is recommended to use this example for complex XDM Schema implementations when you want to easily track the schema version of your implementation and update it with new fields in the future, as well as when you store these objects in memory or serialize/desialized them in your own implementation.
```
Comment (nporter-adbe) [inline docs/ExtensionUsage.md:78]: desialized -> deserialized

File: docs/ExtensionUsage.md [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+Edge.sendEvent(experienceEvent: experienceEvent)
+```
+
+###### Android
+
+```java
+Edge.sendEvent(experienceEvent, null);
+```
+
+By default this API sends the `experienceEvent` to Adobe Experience Edge Network to the Adobe Experience Platform Experience dataset configured in your Edge Configuration.
```
Comment (kevinlind) [inline docs/ExtensionUsage.md:246]: Maybe reword to:
"The `sendEvent` API sends the `experienceEvent` to the Adobe Experience Edge Network. If a dataset ID is not set in the ExperienceEvent, then by default the `experienceEvent` is sent to the dataset configured in the application's Edge Configuration."

File: docs/ExtensionUsage.md [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+  }
+```
+
+
+
+## Edge extension usage for extension developers
+
+**Create Experience Event:**
+
+The XDM data is required for any Experience Event being sent using the AEP Edge mobile extension. Make sure that the data sent in the `xdm` data block, is following the XDM format of a known, publicly available mixin. Along with your XDM data it is encouraged that you set the `eventType` for the requests made by your extension, for easier tracking in the upstream services.
```
Comment (kevinlind) [inline docs/ExtensionUsage.md:291]: > Make sure that the data sent in the `xdm` data block, is following the XDM format of a known, publicly available mixin.

Maybe change to:

> Make sure the `xdm` data block is following the XDM format of a known mixin, either publicly available or defined by, the experience organization used by the application.

File: docs/ExtensionUsage.md [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+}
+```
+
+
+
+**Shared state and extension name**
+
+com.adobe.edge
+
+No data is shared as Shared state by the Edge extension in current version.
```
Comment (kevinlind) [inline docs/ExtensionUsage.md:542]: "in **the** current version"

**Review by kevinlind — COMMENTED** [review]
Looks great! Added some suggestions.

**Review by kevinlind — APPROVED** [review]

**Review by nporter-adbe — APPROVED** [review]
